src/types/types.tsx

```diff
@@ -0,0 +1,32 @@
+export interface Coin {
```

Comment:
I kind of understand the logic of extracting `Coin` & `CoinsList  `interfaces into a common types folder, but they seem better placed in the folders they are closer related to. I suggest placing `Coin `into `CoinTable` folder (separate `.ts` file), and renaming `CoinsList` to `CoinsListResponse `or something similar since it's an API response type, and placing it in a separate file in `api `folder